app/controllers/refinery/blog/admin/posts_controller.rb

```diff
@@ -86,7 +86,8 @@ def post_params
       protected
 
         def find_post
-          @post = Refinery::Blog::Post.find_by_slug_or_id(params[:id])
+          @post = Refinery::Blog::Post.find_by_slug(params[:id]) if params[:id].friendly_id?
+          @post = Refinery::Blog::Post.find(params[:id]) if @post.nil?
```

Comment:
@prokopsimek does @parndt 's version work for you? It looks like the best idea how to solve this problel.